rust/src/dataset/scanner.rs

```diff
@@ -216,24 +216,37 @@ impl Scanner {
     pub fn schema(&self) -> Result<SchemaRef> {
         if self.nearest.as_ref().is_some() {
             let q = self.nearest.as_ref().unwrap();
-            let column: ArrowField = self
-                .dataset
-                .schema()
-                .field(q.column.as_str())
-                .ok_or_else(|| {
-                    Error::Schema(format!("Vector column {} not found in schema", q.column))
-                })?
-                .into();
+            let vector_schema = self.dataset.schema().project(&[&q.column])?;
             let score = ArrowField::new("score", Float32, false);
-            let score_schema = ArrowSchema::new(vec![column, score]);
-            let vector_search_columns = &Schema::try_from(&score_schema)?;
-            let merged = self.projections.merge(vector_search_columns);
+            let score_schema = ArrowSchema::new(vec![score]);
+
+            let merged = self
+                .projections
```

Comment:
what is the differnence of `vector_schema` w/ `self.projections`? is one a superset of the other?

Reply:
=> vector_schema is just the vector and score.
=> self.projections is the user supplied projection (e.g., dataset.to_table(columns=<>, ...), if None then all columns of the dataset is the projection).

Neither is the superset of the other.